main/test/ca/uwaterloo/flix/library/TestMutDisjointSet.flix

```diff
@@ -0,0 +1,389 @@
+        Assert.eq(Some(0), actual)
+    }
+
+    @Test
+    def union08(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(1, 0, s);
+        MutDisjointSet.union(0, 1, s);
+        MutDisjointSet.union(1, 1, s);
+        let actual = MutDisjointSet.find(1, s);
+        Assert.eq(Some(0), actual)
+    }
+
+    @Test
+    def union09(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(0, 1, s);
+        MutDisjointSet.union(1, 3, s);
+        let actual = MutDisjointSet.find(3, s);
+        Assert.eq(Some(1), actual)
+    }
+
+    @Test
+    def union10(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets("a" :: "b" :: "c" :: "d" :: Nil, s);
+        MutDisjointSet.union("c", "d", s);
+        let actual = MutDisjointSet.find("a", s);
+        Assert.eq(Some("a"), actual)
+    }
+
+    @Test
+    def union11(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets("a" :: "b" :: "c" :: "d" :: Nil, s);
+        MutDisjointSet.union("c", "d", s);
+        MutDisjointSet.union("b", "c", s);
+        MutDisjointSet.union("a", "b", s);
+        let actual = MutDisjointSet.find("a", s);
+        Assert.eq(Some("d"), actual)
+    }
+
+    @Test
+    def union12(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(7, 4, s);
+        MutDisjointSet.union(7, 9, s);
+        let actual = MutDisjointSet.find(9, s);
+        Assert.eq(Some(4), actual)
+    }
+
+    @Test
+    def union13(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(7, 4, s);
+        MutDisjointSet.union(7, 9, s);
+        MutDisjointSet.union(8, 1, s);
+        let actual = MutDisjointSet.find(9, s);
+        Assert.eq(Some(4), actual)
+    }
+
+    @Test
+    def union14(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(7, 4, s);
+        MutDisjointSet.union(7, 9, s);
+        MutDisjointSet.union(8, 1, s);
+        MutDisjointSet.union(8, 9, s);
+        let actual = MutDisjointSet.find(8, s);
+        Assert.eq(Some(4), actual)
+    }
+
+    @Test
+    def union15(): Bool = region rc {
+        let s = MutDisjointSet.new(rc);
+        MutDisjointSet.makeSets(List.range(0, 10), s);
+        MutDisjointSet.union(7, 4, s);
+        MutDisjointSet.union(7, 9, s);
+        MutDisjointSet.union(8, 1, s);
+        MutDisjointSet.union(2, 3, s);
+        MutDisjointSet.union(2, 8, s);
+        MutDisjointSet.union(8, 9, s);
+        let actual = MutDisjointSet.find(4, s);
+        Assert.eq(Some(1), actual)
+    }
+
```

Comment:
Do we feel comfortable that we have tested union well enough?

Reply:
We have tests for the following
- Simple call to `union` works
- `union` is idempotent
- `union` on `x, x` does nothing
- `union(x, y)` makes `y` root (both by finding the root element and checking it equals itself, and finding the children (x) and seeing that the returned element is `y`)
- `union(x, y)` and `union(y, x)` leaves `y` as root
- Performing a series of unions that generate two subtrees and taking the union of those subtrees
- Elements that do not exist